src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ArchiveRepoSpecBuilder.java

```diff
@@ -77,6 +83,21 @@ public ArchiveRepoSpecBuilder setRemotePatches(ImmutableMap<String, String> remo
     return this;
   }
 
+  @CanIgnoreReturnValue
+  public ArchiveRepoSpecBuilder setOverlay(ImmutableMap<String, RemoteFile> overlay) {
+    Map<String, List<String>> remoteFiles =
+        overlay.entrySet().stream()
+            .collect(
+                toMap(
```

Comment:
prefer using `ImmutableMap.toImmutableMap` with a static import (same below; and for `ImmutableList.toImmutableList`)